src/components/multi-select/multi-select.jsx

```diff
@@ -0,0 +1,316 @@
+    iconClear: props.readOnly ? styles['icon-clear-readonly'] : styles['icon-clear'],
+    iconsContainer: styles['icon-container'],
+    input: props.readOnly ? styles['input-readonly'] : styles.input,
+    noOptionsContainer: styles['no-options'],
+    popupContainer: styles['popup-container'],
+    tagList: styles['tag-list'],
+    ...props.classNames,
+  };
+
+  const dropdownContents = () => {
+    if (!expanded) {
+      return undefined;
+    }
+
+    if (props.showLoader) {
+      return (
+        <Listbox
+          className={classNames.popupContainer}
+          id={ids.listbox}
+          labelledBy={ids.label}
+          refs={[]}
+        >
+          <Loader inline />
+        </Listbox>
+      );
+    }
+
+    if (visibleOptions.length === 0) {
+      return (<NoOptions className={classNames.noOptionsContainer} id={ids.emptyMessage} />);
+    }
+
+    return (
+      <Listbox
+        className={classNames.popupContainer}
+        id={ids.listbox}
+        labelledBy={ids.label}
+        refs={visibleOptionsRefs}
+      >
+        {props.listboxChildren ?
+          props.listboxChildren(visibleOptions, visibleOptionsRefs, onOptionSelect) :
+          visibleOptions.map((option, index) => (
+            <ListOption
+              key={`${props.id}-${props.optionIDGetter(option)}`}
+              onSelect={onOptionSelect}
+              ref={visibleOptionsRefs[index]}
+              value={props.optionIDGetter(option)}
+            >
+              {props.optionLabelGetter(option)}
+            </ListOption>
+          ))}
+      </Listbox>
+    );
+  };
+
+  function getOption(id) {
+    return props.options.find(option => props.optionIDGetter(option) === id);
+  }
+
+  function getVisibleOptions() {
+    return props.options
+      .filter(option => props.optionLabelGetter(option).toLowerCase().includes(autocompleteValue.toLowerCase()))
+      .filter(option => !value.some((val) => {
+        const optionID = props.optionIDGetter(option);
+
+        if (typeof optionID === 'string') {
+          return val.toLowerCase() === optionID.toLowerCase();
+        }
+
+        return val === optionID;
+      }));
+  }
+
+  function onAutocompleteBlur() {
+    if (!autocompleteRef.current) return;
+
+    autocompleteRef.current.setCustomValidity('');
+    if (!autocompleteRef.current.validity.valid) {
+      setValidationMessage(autocompleteRef.current.validationMessage);
+    } else {
+      setValidationMessage(props.validationMessage || '');
+    }
+  }
+
+  function onAutocompleteChange(event) {
+    setExpanded(true);
+    setAutocompleteValue(event.target.value);
+  }
+
+  function onDropdownClose() {
+    setExpanded(false);
+    setAutocompleteValue('');
+  }
+
+  function onOptionRemove(event) {
+    const newValue = value.filter((option, index) => (
+      valueRefs[index].current !== event.target
+    ));
+    setValue(newValue);
+  }
+
+  function onOptionsClear(event) {
+    event.preventDefault();
```

Comment:
Note: Do we always want to prevent the icon click from propagating? That was my assumption, but this can be moved inside the `if` block below otherwise.

Reply:
No, I think we still need the event to propagate in case there are window click event handlers. But I do like the idea of we "prevent the default" of opening the popup.